### PR TITLE
Font Library: address JS feedback in #57688

### PR DESCRIPTION
**Pull request by mikachan**
## What?

Addresses JS feedback in https://github.com/WordPress/gutenberg/pull/57688.

## Testing Instructions

Ensure installing and deleting font families still works without error using the Font Library UI.

### CODE REVIEW

File: packages/edit-site/src/components/global-styles/font-library-modal/resolvers.js [hunk 1]
```diff
@@ -7,76 +7,76 @@
...
+	return {
+		id: response.id,
+		...response.font_face_settings,
+	};
 }
 
 export async function fetchGetFontFamilyBySlug( slug ) {
 	const config = {
-		path: `/wp/v2/font-families?slug=${ slug }&_embed=true`,
+		path: `${ FONT_FAMILIES_URL }s?slug=${ slug }&_embed=true`,
```
Comment (jffng) [inline packages/edit-site/src/components/global-styles/font-library-modal/resolvers.js]:
```suggestion
		path: `${ FONT_FAMILIES_URL }?slug=${ slug }&_embed=true`,
```

**Review by jffng — COMMENTED** [review]
This still works as expected, but I'm wondering why adding `await` is necessary, since how these functions are invoked doesn't change at all.

**Comment by pbking** [comment]
> I'm wondering why adding await is necessary

If the function isn't noted as being async (via 'await') then the return will happen before the fetch is complete.  `response` will be a Promise object rather than the object the promise resolves to.

Prior to this change we were using the `.then()` syntax to accomplish the same thing, but was was being returned from the function was a promise.  It was, however, returned immediately so the function wasn't an async one.  The PROMISE that was returned resolved async, but the returning of that promise wasn't.

That's why mixing the two wasn't correct.  We said the function was async and it wasn't.

Now it is, but to do that we have to await the fetch call and (later) return our result of processing that response.

**Review by pbking — APPROVED** [review]
These changes look good to me and the tooling continues to work as expected.

**Pull request merged**